### PR TITLE
bug 1482159: Get sample server from config for LiveSampleURL

### PR DESCRIPTION
**Pull request by jwhitlock**
Instead of guessing the sample server domain and protocol from the requesting domain, add a new config item ``live_samples.base_url`` and use it to construct the sample URL, using the [WHATWG URL API](https://nodejs.org/api/url.html#url_the_whatwg_url_api). Plus tests!

I need to use a different sample server domain in local dev to reproduce the issue in [bug 1482159](https://bugzilla.mozilla.org/show_bug.cgi?id=1482159).

### CODE REVIEW

**Comment by jwhitlock** [comment]
Force-push to get ``{{EmbedLiveSample}}`` tests from PR #785, and to add another test for the change to unicode IDs, which are now URL encoded.

File: tests/macros/test-EmbedLiveSample.js [hunk 1]
```diff
@@ -11,6 +11,7 @@ chai.use(chaiAsPromised);
 
 describeMacro('EmbedLiveSample', function () {
     itMacro('One argument: ID', function (macro) {
+        macro.ctx.env.live_samples = {'base_url': 'https://mdn.mozillademos.org'};
```
[the remark this answers is not in the document]
Comment (jwhitlock) [reply tests/macros/test-EmbedLiveSample.js]: Nice, I wasn't aware of that feature!
